TestFileStoreOnRookInstalledViaHelm failing intermittently

We are seeing intermittent failures in the TestFileStoreOnRookInstalledViaHelm test recently.
```
--- FAIL: TestFileStoreOnRookInstalledViaHelm (17.33s)
	assertions.go:229:                         
	Error Trace:	base_file_test.go:90		
			helm_test.go:84		
	Error:		Not equal: 
			expected: 1
			received: 2		
	Messages:	There should one shared file system 
```

A couple builds that had this behavior:
https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/master/300/pipeline/37/
https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/PR-1077/19/pipeline/

--- a/tests/framework/installer/install_data.go
+++ b/tests/framework/installer/install_data.go
@@ -16,8 +16,6 @@ limitations under the License.
 
 package installer
 
-import "strings"
-
 //InstallData wraps rook yaml definitions
 type InstallData struct {
 }
@@ -28,55 +26,12 @@ func NewK8sInstallData() *InstallData {
 }
 
 //GetRookOperator returns rook Operator  manifest
-func (i *InstallData) GetRookOperator(k8sVersion string) string {
-
-	if strings.Contains(k8sVersion, "v1.5") {
-		return `kind: Namespace
-apiVersion: v1
-metadata:
-  name: rook-system
----
-apiVersion: extensions/v1beta1
-kind: Deployment
-metadata:
-  name: rook-operator
-  namespace: rook-system
-spec:
-  replicas: 1
-  template:
-    metadata:
-      labels:
-        name: rook-operator
-    spec:
-      containers:
-      - name: rook-operator
-        image: rook/rook:master
-        args: ["operator", "--mon-healthcheck-interval=5s", "--mon-out-timeout=1s"]
-        env:
-        - name: ROOK_REPO_PREFIX
-          value: rook
-        - name: NODE_NAME
-          valueFrom:
-            fieldRef:
-              fieldPath: spec.nodeName
-        - name: ROOK_OPERATOR_SERVICE_ACCOUNT
-          valueFrom:
-            fieldRef:
-              fieldPath: spec.serviceAccountName
-        - name: POD_NAME
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.name
-        - name: POD_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace`
-	}
+func (i *InstallData) GetRookOperator(k8sVersion, namespace string) string {
 
 	return `kind: Namespace
 apiVersion: v1
 metadata:
-  name: rook-system
+  name: ` + namespace + `
 ---
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1beta1
@@ -160,13 +115,13 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: rook-operator
-  namespace: rook-system
+  namespace: ` + namespace + `
 ---
 kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1beta1
 metadata:
   name: rook-operator
-  namespace: rook-system
+  namespace: ` + namespace + `
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: ClusterRole
@@ -174,13 +129,13 @@ roleRef:
 subjects:
 - kind: ServiceAccount
   name: rook-operator
-  namespace: rook-system
+  namespace: ` + namespace + `
 ---
 apiVersion: apps/v1beta1
 kind: Deployment
 metadata:
   name: rook-operator
-  namespace: rook-system
+  namespace: ` + namespace + `
 spec:
   replicas: 1
   template:

--- a/tests/framework/installer/install_helper.go
+++ b/tests/framework/installer/install_helper.go
@@ -58,12 +58,12 @@ func init() {
 }
 
 //CreateK8sRookOperator creates rook-operator via kubectl
-func (h *InstallHelper) CreateK8sRookOperator() (err error) {
+func (h *InstallHelper) CreateK8sRookOperator(namespace string) (err error) {
 	logger.Infof("Starting Rook Operator")
 	//creating clusterrolebinding for kubeadm env.
 	h.k8shelper.CreateAnonSystemClusterBinding()
 
-	rookOperator := h.installData.GetRookOperator(h.k8shelper.GetK8sServerVersion())
+	rookOperator := h.installData.GetRookOperator(h.k8shelper.GetK8sServerVersion(), namespace)
 
 	_, err = h.k8shelper.KubectlWithStdin(rookOperator, createArgs...)
 
@@ -117,10 +117,10 @@ func (h *InstallHelper) CreateK8sRookOperatorViaHelm(namespace string) (err erro
 }
 
 //CreateK8sRookToolbox creates rook-tools via kubectl
-func (h *InstallHelper) CreateK8sRookToolbox(clusterNamespace string) (err error) {
+func (h *InstallHelper) CreateK8sRookToolbox(namespace string) (err error) {
 	logger.Infof("Starting Rook toolbox")
 
-	rookToolbox := h.installData.GetRookToolBox(clusterNamespace)
+	rookToolbox := h.installData.GetRookToolBox(namespace)
 
 	_, err = h.k8shelper.KubectlWithStdin(rookToolbox, createArgs...)
 
@@ -128,7 +128,7 @@ func (h *InstallHelper) CreateK8sRookToolbox(clusterNamespace string) (err error
 		return fmt.Errorf("Failed to create rook-toolbox pod : %v ", err)
 	}
 
-	if !h.k8shelper.IsPodRunning("rook-tools", clusterNamespace) {
+	if !h.k8shelper.IsPodRunning("rook-tools", namespace) {
 		return fmt.Errorf("Rook Toolbox couldn't start")
 	}
 	logger.Infof("Rook Toolbox started")
@@ -137,10 +137,10 @@ func (h *InstallHelper) CreateK8sRookToolbox(clusterNamespace string) (err error
 }
 
 //CreateK8sRookCluster creates rook cluster via kubectl
-func (h *InstallHelper) CreateK8sRookCluster(clusterNamespace string) (err error) {
+func (h *InstallHelper) CreateK8sRookCluster(namespace string) (err error) {
 	logger.Infof("Starting Rook Cluster")
 
-	rookCluster := h.installData.GetRookCluster(clusterNamespace)
+	rookCluster := h.installData.GetRookCluster(namespace)
 
 	_, err = h.k8shelper.KubectlWithStdin(rookCluster, createArgs...)
 
@@ -148,7 +148,7 @@ func (h *InstallHelper) CreateK8sRookCluster(clusterNamespace string) (err error
 		return fmt.Errorf("Failed to create rook cluster : %v ", err)
 	}
 
-	if !h.k8shelper.IsServiceUp("rook-api", clusterNamespace) {
+	if !h.k8shelper.IsServiceUp("rook-api", namespace) {
 		logger.Infof("Rook Cluster couldn't start")
 	} else {
 		logger.Infof("Rook Cluster started")
@@ -157,8 +157,12 @@ func (h *InstallHelper) CreateK8sRookCluster(clusterNamespace string) (err error
 	return nil
 }
 
+func SystemNamespace(namespace string) string {
+	return fmt.Sprintf("%s-system", namespace)
+}
+
 //InstallRookOnK8s installs rook on k8s
-func (h *InstallHelper) InstallRookOnK8s(clusterNamespace string) (err error) {
+func (h *InstallHelper) InstallRookOnK8s(namespace string) (err error) {
 
 	//flag used for local debuggin purpose, when rook is pre-installed
 	skipRookInstall := strings.EqualFold(h.Env.SkipInstallRook, "true")
@@ -174,28 +178,28 @@ func (h *InstallHelper) InstallRookOnK8s(clusterNamespace string) (err error) {
 		panic(err)
 	}
 
-	err = h.CreateK8sRookOperator()
+	err = h.CreateK8sRookOperator(SystemNamespace(namespace))
 	if err != nil {
 		fmt.Println(err)
 		panic(err)
 	}
 
-	if !h.k8shelper.IsPodInExpectedState("rook-operator", "rook-system", "Running") {
+	if !h.k8shelper.IsPodInExpectedState("rook-operator", SystemNamespace(namespace), "Running") {
 		fmt.Println("rook-operator is not running")
-		h.k8shelper.GetRookLogs("rook-operator", "rook-system", "test-setup")
+		h.k8shelper.GetRookLogs("rook-operator", SystemNamespace(namespace), "test-setup")
 		panic("rook-operator is not Running, abort!")
 	}
 
-	if !h.k8shelper.IsPodInExpectedState("rook-agent", "rook-system", "Running") {
+	if !h.k8shelper.IsPodInExpectedState("rook-agent", SystemNamespace(namespace), "Running") {
 		fmt.Println("rook-agent is not running")
-		h.k8shelper.GetRookLogs("rook-agent", "rook-system", "test-setup")
+		h.k8shelper.GetRookLogs("rook-agent", SystemNamespace(namespace), "test-setup")
 		panic("rook-agent is not Running, abort!")
 	}
 
 	time.Sleep(10 * time.Second)
 
 	//Create rook cluster
-	err = h.CreateK8sRookCluster(clusterNamespace)
+	err = h.CreateK8sRookCluster(namespace)
 	if err != nil {
 		panic(err)
 	}
@@ -203,16 +207,16 @@ func (h *InstallHelper) InstallRookOnK8s(clusterNamespace string) (err error) {
 	time.Sleep(5 * time.Second)
 
 	//Create rook client
-	err = h.CreateK8sRookToolbox(clusterNamespace)
+	err = h.CreateK8sRookToolbox(namespace)
 	if err != nil {
 		panic(err)
 	}
-	logger.Infof("installed rook operator and cluster : %s on k8s %s", clusterNamespace, h.Env.K8sVersion)
+	logger.Infof("installed rook operator and cluster : %s on k8s %s", namespace, h.Env.K8sVersion)
 	return nil
 }
 
 //UninstallRookFromK8s uninstalls rook from k8s
-func (h *InstallHelper) UninstallRookFromK8s(clusterNamespace string, helmInstalled bool) {
+func (h *InstallHelper) UninstallRookFromK8s(namespace string, helmInstalled bool) {
 	//flag used for local debugging purpose, when rook is pre-installed
 	skipRookInstall := strings.EqualFold(h.Env.SkipInstallRook, "true")
 	if skipRookInstall {
@@ -230,7 +234,7 @@ func (h *InstallHelper) UninstallRookFromK8s(clusterNamespace string, helmInstal
 			panic(err)
 		}
 	} else {
-		rookOperator := h.installData.GetRookOperator(h.k8shelper.GetK8sServerVersion())
+		rookOperator := h.installData.GetRookOperator(h.k8shelper.GetK8sServerVersion(), SystemNamespace(namespace))
 
 		_, err = h.k8shelper.KubectlWithStdin(rookOperator, deleteArgs...)
 		if err != nil {
@@ -238,15 +242,15 @@ func (h *InstallHelper) UninstallRookFromK8s(clusterNamespace string, helmInstal
 		}
 		k8sHelp.WaitUntilNameSpaceIsDeleted("rook-system")
 	}
-	_, err = k8sHelp.DeleteResource([]string{"-n", clusterNamespace, "cluster", clusterNamespace})
+	_, err = k8sHelp.DeleteResource([]string{"-n", namespace, "cluster", namespace})
 	if err != nil {
 		panic(err)
 	}
-	_, err = k8sHelp.DeleteResource([]string{"-n", clusterNamespace, "serviceaccount", "rook-api"})
+	_, err = k8sHelp.DeleteResource([]string{"-n", namespace, "serviceaccount", "rook-api"})
 	if err != nil {
 		panic(err)
 	}
-	_, err = k8sHelp.DeleteResource([]string{"-n", clusterNamespace, "serviceaccount", "rook-ceph-osd"})
+	_, err = k8sHelp.DeleteResource([]string{"-n", namespace, "serviceaccount", "rook-ceph-osd"})
 	if err != nil {
 		panic(err)
 	}
@@ -274,20 +278,20 @@ func (h *InstallHelper) UninstallRookFromK8s(clusterNamespace string, helmInstal
 			panic(err)
 		}
 	}
-	_, err = k8sHelp.DeleteResource([]string{"namespace", clusterNamespace})
+	_, err = k8sHelp.DeleteResource([]string{"namespace", namespace})
 	if err != nil {
 		panic(err)
 	}
 
-	isRookUninstalled := k8sHelp.WaitUntilPodInNamespaceIsDeleted("rook-ceph-mon", clusterNamespace)
-	isNameSpaceDeleted := k8sHelp.WaitUntilNameSpaceIsDeleted(clusterNamespace)
+	isRookUninstalled := k8sHelp.WaitUntilPodInNamespaceIsDeleted("rook-ceph-mon", namespace)
+	isNameSpaceDeleted := k8sHelp.WaitUntilNameSpaceIsDeleted(namespace)
 	h.k8shelper.Clientset.RbacV1beta1().ClusterRoleBindings().Delete("anon-user-access", nil)
 
 	if isRookUninstalled && isNameSpaceDeleted {
-		logger.Infof("Rook cluster %s uninstalled successfully", clusterNamespace)
+		logger.Infof("Rook cluster %s uninstalled successfully", namespace)
 		return
 	}
-	panic(fmt.Errorf("Rook cluster %s not uninstalled", clusterNamespace))
+	panic(fmt.Errorf("Rook cluster %s not uninstalled", namespace))
 }
 
 //CleanupCluster deletes a rook cluster for a namespace

--- a/tests/integration/base_deploy_test.go
+++ b/tests/integration/base_deploy_test.go
@@ -26,11 +26,7 @@ import (
 var (
 	logger = capnslog.NewPackageLogger("github.com/rook/rook", "integrationTest")
 
-	defaultNamespace           = "default"
-	defaultRookSystemNamespace = "rook-system"
-	defaultRookNamespace       = "rook"
-	clusterNamespace1          = "test-cluster-1"
-	clusterNamespace2          = "test-cluster-2"
+	defaultNamespace = "default"
 )
 
 //Test to make sure all rook components are installed and Running

--- a/tests/integration/base_file_test.go
+++ b/tests/integration/base_file_test.go
@@ -87,6 +87,7 @@ func runFileE2ETestLite(helper *clients.TestClient, k8sh *utils.K8sHelper, s sui
 	fscErr := fc.FSCreate(filesystemName, namespace, true, k8sh)
 	require.Nil(s.T(), fscErr)
 	filesystemList, _ := fc.FSList()
+	logger.Infof("Retrieved file systems: %+v", filesystemList)
 	require.Equal(s.T(), 1, len(filesystemList), "There should one shared file system present")
 }
 

--- a/tests/integration/helm_test.go
+++ b/tests/integration/helm_test.go
@@ -28,9 +28,11 @@ type HelmSuite struct {
 	k8sh      *utils.K8sHelper
 	installer *installer.InstallHelper
 	hh        *utils.HelmHelper
+	namespace string
 }
 
 func (hs *HelmSuite) SetupSuite() {
+	hs.namespace = "helm-ns"
 	kh, err := utils.CreateK8sHelper(hs.T)
 	require.NoError(hs.T(), err)
 
@@ -39,54 +41,54 @@ func (hs *HelmSuite) SetupSuite() {
 
 	hs.installer = installer.NewK8sRookhelper(kh.Clientset, hs.T)
 
-	err = hs.installer.CreateK8sRookOperatorViaHelm(defaultRookNamespace)
+	err = hs.installer.CreateK8sRookOperatorViaHelm(hs.namespace)
 	require.NoError(hs.T(), err)
 
-	require.True(hs.T(), kh.IsPodInExpectedState("rook-operator", defaultRookNamespace, "Running"),
+	require.True(hs.T(), kh.IsPodInExpectedState("rook-operator", hs.namespace, "Running"),
 		"Make sure rook-operator is in running state")
 
-	require.True(hs.T(), kh.IsPodInExpectedState("rook-agent", defaultRookNamespace, "Running"),
+	require.True(hs.T(), kh.IsPodInExpectedState("rook-agent", hs.namespace, "Running"),
 		"Make sure rook-agent is in running state")
 
 	time.Sleep(10 * time.Second)
 
-	err = hs.installer.CreateK8sRookCluster(defaultRookNamespace)
+	err = hs.installer.CreateK8sRookCluster(hs.namespace)
 	require.NoError(hs.T(), err)
 
-	err = hs.installer.CreateK8sRookToolbox(defaultRookNamespace)
+	err = hs.installer.CreateK8sRookToolbox(hs.namespace)
 	require.NoError(hs.T(), err)
 
-	hs.helper, err = clients.CreateTestClient(enums.Kubernetes, kh, defaultRookNamespace)
+	hs.helper, err = clients.CreateTestClient(enums.Kubernetes, kh, hs.namespace)
 	require.Nil(hs.T(), err)
 }
 
 func (hs *HelmSuite) TearDownSuite() {
 	if hs.T().Failed() {
-		gatherAllRookLogs(hs.k8sh, hs.Suite, defaultRookNamespace, defaultRookNamespace)
+		gatherAllRookLogs(hs.k8sh, hs.Suite, hs.namespace, hs.namespace)
 	}
-	hs.installer.UninstallRookFromK8s(defaultRookNamespace, true)
+	hs.installer.UninstallRookFromK8s(hs.namespace, true)
 
 }
 
 //Test to make sure all rook components are installed and Running
 func (hs *HelmSuite) TestRookInstallViaHelm() {
-	checkIfRookClusterIsInstalled(hs.k8sh, hs.Suite, defaultRookNamespace, defaultRookNamespace)
+	checkIfRookClusterIsInstalled(hs.k8sh, hs.Suite, hs.namespace, hs.namespace)
 
 }
 
 //Test BlockCreation on Rook that was installed via Helm
 func (hs *HelmSuite) TestBlockStoreOnRookInstalledViaHelm() {
-	runBlockE2ETestLite(hs.helper, hs.k8sh, hs.Suite, defaultRookNamespace)
+	runBlockE2ETestLite(hs.helper, hs.k8sh, hs.Suite, hs.namespace)
 }
 
 //Test File System Creation on Rook that was installed via helm
 func (hs *HelmSuite) TestFileStoreOnRookInstalledViaHelm() {
-	runFileE2ETestLite(hs.helper, hs.k8sh, hs.Suite, defaultRookNamespace, "testfs")
+	runFileE2ETestLite(hs.helper, hs.k8sh, hs.Suite, hs.namespace, "testfs")
 
 }
 
 //Test Object StoreCreation on Rook that was installed via helm
 func (hs *HelmSuite) TestObjectStoreOnRookInstalledViaHelm() {
-	runObjectE2ETestLite(hs.helper, hs.k8sh, hs.Suite, defaultRookNamespace, "default", 3)
+	runObjectE2ETestLite(hs.helper, hs.k8sh, hs.Suite, hs.namespace, "default", 3)
 
 }

--- a/tests/integration/mon_test.go
+++ b/tests/integration/mon_test.go
@@ -30,7 +30,7 @@ func (suite *SmokeSuite) TestMonFailover() {
 	logger.Infof("Mon Failover Smoke Test")
 
 	opts := metav1.ListOptions{LabelSelector: "app=rook-ceph-mon"}
-	monReplicas, err := suite.k8sh.Clientset.ReplicaSets(defaultRookNamespace).List(opts)
+	monReplicas, err := suite.k8sh.Clientset.ReplicaSets(suite.namespace).List(opts)
 	require.Nil(suite.T(), err)
 	assert.Equal(suite.T(), 3, len(monReplicas.Items))
 
@@ -38,12 +38,12 @@ func (suite *SmokeSuite) TestMonFailover() {
 	logger.Infof("Killing mon %s", monToKill)
 	propagation := metav1.DeletePropagationForeground
 	delOptions := &metav1.DeleteOptions{PropagationPolicy: &propagation}
-	err = suite.k8sh.Clientset.ReplicaSets(defaultRookNamespace).Delete(monToKill, delOptions)
+	err = suite.k8sh.Clientset.ReplicaSets(suite.namespace).Delete(monToKill, delOptions)
 	require.Nil(suite.T(), err)
 
 	// Wait for the health check to start a new monitor
 	for i := 0; i < 10; i++ {
-		monReplicas, err := suite.k8sh.Clientset.ReplicaSets(defaultRookNamespace).List(opts)
+		monReplicas, err := suite.k8sh.Clientset.ReplicaSets(suite.namespace).List(opts)
 		require.Nil(suite.T(), err)
 
 		// Make sure the old mon is not still alive

--- a/tests/integration/multi_cluster_deploy_test.go
+++ b/tests/integration/multi_cluster_deploy_test.go
@@ -24,6 +24,8 @@ type MultiRookClusterDeploySuite struct {
 	k8sh        *utils.K8sHelper
 	installer   *installer.InstallHelper
 	installData *installer.InstallData
+	namespace1  string
+	namespace2  string
 }
 
 //Deploy Multiple Rook clusters
@@ -37,41 +39,41 @@ func (mrc *MultiRookClusterDeploySuite) SetupSuite() {
 	mrc.installer = installer.NewK8sRookhelper(kh.Clientset, mrc.T)
 	mrc.installData = installer.NewK8sInstallData()
 
-	err = mrc.installer.CreateK8sRookOperator()
+	err = mrc.installer.CreateK8sRookOperator(installer.SystemNamespace(mrc.namespace1))
 	require.NoError(mrc.T(), err)
 
-	require.True(mrc.T(), kh.IsPodInExpectedState("rook-operator", defaultRookSystemNamespace, "Running"),
+	require.True(mrc.T(), kh.IsPodInExpectedState("rook-operator", mrc.namespace1, "Running"),
 		"Make sure rook-operator is in running state")
 
-	require.True(mrc.T(), kh.IsPodInExpectedState("rook-agent", defaultRookSystemNamespace, "Running"),
+	require.True(mrc.T(), kh.IsPodInExpectedState("rook-agent", mrc.namespace1, "Running"),
 		"Make sure rook-agent is in running state")
 
 	time.Sleep(10 * time.Second)
 
-	err = mrc.installer.CreateK8sRookCluster(clusterNamespace1)
+	err = mrc.installer.CreateK8sRookCluster(mrc.namespace1)
 	require.NoError(mrc.T(), err)
 
-	err = mrc.installer.CreateK8sRookToolbox(clusterNamespace1)
+	err = mrc.installer.CreateK8sRookToolbox(mrc.namespace1)
 	require.NoError(mrc.T(), err)
 
-	err = mrc.installer.CreateK8sRookCluster(clusterNamespace2)
+	err = mrc.installer.CreateK8sRookCluster(mrc.namespace2)
 	require.NoError(mrc.T(), err)
 
-	err = mrc.installer.CreateK8sRookToolbox(clusterNamespace2)
+	err = mrc.installer.CreateK8sRookToolbox(mrc.namespace2)
 	require.NoError(mrc.T(), err)
 
-	mrc.helper1, err = clients.CreateTestClient(enums.Kubernetes, kh, clusterNamespace1)
+	mrc.helper1, err = clients.CreateTestClient(enums.Kubernetes, kh, mrc.namespace1)
 	require.Nil(mrc.T(), err)
 
-	mrc.helper2, err = clients.CreateTestClient(enums.Kubernetes, kh, clusterNamespace2)
+	mrc.helper2, err = clients.CreateTestClient(enums.Kubernetes, kh, mrc.namespace2)
 	require.Nil(mrc.T(), err)
 
 }
 
 func (mrc *MultiRookClusterDeploySuite) TearDownSuite() {
 	if mrc.T().Failed() {
-		gatherAllRookLogs(mrc.k8sh, mrc.Suite, defaultRookSystemNamespace, clusterNamespace1)
-		gatherAllRookLogs(mrc.k8sh, mrc.Suite, defaultRookSystemNamespace, clusterNamespace2)
+		gatherAllRookLogs(mrc.k8sh, mrc.Suite, installer.SystemNamespace(mrc.namespace1), mrc.namespace1)
+		gatherAllRookLogs(mrc.k8sh, mrc.Suite, installer.SystemNamespace(mrc.namespace1), mrc.namespace2)
 	}
 	deleteArgs := []string{"delete", "-f", "-"}
 
@@ -85,7 +87,7 @@ func (mrc *MultiRookClusterDeploySuite) TearDownSuite() {
 	if err != nil {
 		panic(err)
 	}
-	rookOperator := mrc.installData.GetRookOperator(mrc.k8sh.GetK8sServerVersion())
+	rookOperator := mrc.installData.GetRookOperator(mrc.k8sh.GetK8sServerVersion(), installer.SystemNamespace(mrc.namespace1))
 
 	//Delete rook operator
 	_, err = mrc.k8sh.KubectlWithStdin(rookOperator, deleteArgs...)
@@ -95,8 +97,8 @@ func (mrc *MultiRookClusterDeploySuite) TearDownSuite() {
 	}
 
 	//delete rook cluster
-	mrc.installer.CleanupCluster(clusterNamespace1)
-	mrc.installer.CleanupCluster(clusterNamespace2)
+	mrc.installer.CleanupCluster(mrc.namespace1)
+	mrc.installer.CleanupCluster(mrc.namespace2)
 
 	//Delete ClusterRole and ClusterRoleBindings
 	if mrc.k8sh.VersionAtLeast("v1.6.0") {
@@ -127,17 +129,17 @@ func (mrc *MultiRookClusterDeploySuite) TearDownSuite() {
 		}
 	}
 
-	isRookUninstalled1 := k8sHelp.WaitUntilPodInNamespaceIsDeleted("rook-ceph-mon", clusterNamespace1)
-	isNameSpaceDeleted1 := k8sHelp.WaitUntilNameSpaceIsDeleted(clusterNamespace1)
-	isRookUninstalled2 := k8sHelp.WaitUntilPodInNamespaceIsDeleted("rook-ceph-mon", clusterNamespace2)
-	isNameSpaceDeleted2 := k8sHelp.WaitUntilNameSpaceIsDeleted(clusterNamespace2)
+	isRookUninstalled1 := k8sHelp.WaitUntilPodInNamespaceIsDeleted("rook-ceph-mon", mrc.namespace1)
+	isNameSpaceDeleted1 := k8sHelp.WaitUntilNameSpaceIsDeleted(mrc.namespace1)
+	isRookUninstalled2 := k8sHelp.WaitUntilPodInNamespaceIsDeleted("rook-ceph-mon", mrc.namespace2)
+	isNameSpaceDeleted2 := k8sHelp.WaitUntilNameSpaceIsDeleted(mrc.namespace2)
 	mrc.k8sh.Clientset.RbacV1beta1().ClusterRoleBindings().Delete("anon-user-access", nil)
 
 	if isRookUninstalled1 && isNameSpaceDeleted1 && isRookUninstalled2 && isNameSpaceDeleted2 {
-		logger.Infof("Rook clusters %s  and  %s uninstalled successfully", clusterNamespace1, clusterNamespace2)
+		logger.Infof("Rook clusters %s  and  %s uninstalled successfully", mrc.namespace1, mrc.namespace2)
 		return
 	}
-	logger.Infof("Rook clusters %s  and  %s  not uninstalled successfully", clusterNamespace1, clusterNamespace2)
+	logger.Infof("Rook clusters %s  and  %s  not uninstalled successfully", mrc.namespace1, mrc.namespace2)
 
 }
 
@@ -145,32 +147,32 @@ func (mrc *MultiRookClusterDeploySuite) TearDownSuite() {
 func (mrc *MultiRookClusterDeploySuite) TestInstallingMultipleRookClusters() {
 
 	//Check if Rook cluster 1 is deployed successfully
-	checkIfRookClusterIsInstalled(mrc.k8sh, mrc.Suite, defaultRookSystemNamespace, clusterNamespace1)
+	checkIfRookClusterIsInstalled(mrc.k8sh, mrc.Suite, installer.SystemNamespace(mrc.namespace1), mrc.namespace1)
 
-	//Check if Rook cluster 1 is deployed successfully
-	checkIfRookClusterIsInstalled(mrc.k8sh, mrc.Suite, defaultRookSystemNamespace, clusterNamespace2)
+	//Check if Rook cluster 2 is deployed successfully
+	checkIfRookClusterIsInstalled(mrc.k8sh, mrc.Suite, installer.SystemNamespace(mrc.namespace1), mrc.namespace2)
 
 }
 
 //Test Block Store Creation on multiple rook clusters
 func (mrc *MultiRookClusterDeploySuite) TestBlockStoreOnMultipleRookCluster() {
-	runBlockE2ETestLite(mrc.helper1, mrc.k8sh, mrc.Suite, clusterNamespace1)
-	runBlockE2ETestLite(mrc.helper2, mrc.k8sh, mrc.Suite, clusterNamespace2)
+	runBlockE2ETestLite(mrc.helper1, mrc.k8sh, mrc.Suite, mrc.namespace1)
+	runBlockE2ETestLite(mrc.helper2, mrc.k8sh, mrc.Suite, mrc.namespace2)
 
 }
 
 //Test Filesystem Creation on multiple rook clusters
 func (mrc *MultiRookClusterDeploySuite) TestFileStoreOnMultiRookCluster() {
-	runFileE2ETestLite(mrc.helper1, mrc.k8sh, mrc.Suite, clusterNamespace1, "test-fs-1")
+	runFileE2ETestLite(mrc.helper1, mrc.k8sh, mrc.Suite, mrc.namespace1, "test-fs-1")
 	//TODO - Known Issues #https://github.com/rook/rook/issues/970
-	//runFileE2ETestLite(mrc.helper2, mrc.k8sh, mrc.Suite, clusterNamespace2, "test-fs-2")
+	//runFileE2ETestLite(mrc.helper2, mrc.k8sh, mrc.Suite, mrc.namespace2, "test-fs-2")
 
 }
 
 //Test Object Store Creation on multiple rook clusters
 func (mrc *MultiRookClusterDeploySuite) TestObjectStoreOnMultiRookCluster() {
-	runObjectE2ETestLite(mrc.helper1, mrc.k8sh, mrc.Suite, clusterNamespace1, "default-c1", 2)
+	runObjectE2ETestLite(mrc.helper1, mrc.k8sh, mrc.Suite, mrc.namespace1, "default-c1", 2)
 	//TODO - Known Issues #https://github.com/rook/rook/issues/970
-	//runObjectE2ETestLite(mrc.helper2, mrc.k8sh, mrc.Suite, clusterNamespace2, "default-c2", 1)
+	//runObjectE2ETestLite(mrc.helper2, mrc.k8sh, mrc.Suite, mrc.namespace2, "default-c2", 1)
 
 }

--- a/tests/integration/smoke_test.go
+++ b/tests/integration/smoke_test.go
@@ -36,9 +36,11 @@ type SmokeSuite struct {
 	helper    *clients.TestClient
 	k8sh      *utils.K8sHelper
 	installer *installer.InstallHelper
+	namespace string
 }
 
 func (suite *SmokeSuite) SetupSuite() {
+	suite.namespace = "smoke-ns"
 	kh, err := utils.CreateK8sHelper(suite.T)
 	require.NoError(suite.T(), err)
 
@@ -46,32 +48,32 @@ func (suite *SmokeSuite) SetupSuite() {
 
 	suite.installer = installer.NewK8sRookhelper(kh.Clientset, suite.T)
 
-	err = suite.installer.InstallRookOnK8s(defaultRookNamespace)
+	err = suite.installer.InstallRookOnK8s(suite.namespace)
 	require.NoError(suite.T(), err)
 
-	suite.helper, err = clients.CreateTestClient(enums.Kubernetes, kh, defaultRookNamespace)
+	suite.helper, err = clients.CreateTestClient(enums.Kubernetes, kh, suite.namespace)
 	require.Nil(suite.T(), err)
 }
 
 func (suite *SmokeSuite) TearDownSuite() {
 	if suite.T().Failed() {
-		gatherAllRookLogs(suite.k8sh, suite.Suite, defaultRookSystemNamespace, defaultRookNamespace)
+		gatherAllRookLogs(suite.k8sh, suite.Suite, installer.SystemNamespace(suite.namespace), suite.namespace)
 
 	}
-	suite.installer.UninstallRookFromK8s(defaultRookNamespace, false)
+	suite.installer.UninstallRookFromK8s(suite.namespace, false)
 }
 
 func (suite *SmokeSuite) TestBlockStorage_SmokeTest() {
-	runBlockE2ETest(suite.helper, suite.k8sh, suite.Suite, defaultRookNamespace)
+	runBlockE2ETest(suite.helper, suite.k8sh, suite.Suite, suite.namespace)
 }
 func (suite *SmokeSuite) TestFileStorage_SmokeTest() {
-	runFileE2ETest(suite.helper, suite.k8sh, suite.Suite, defaultRookNamespace, "smoke-test-fs")
+	runFileE2ETest(suite.helper, suite.k8sh, suite.Suite, suite.namespace, "smoke-test-fs")
 }
 func (suite *SmokeSuite) TestObjectStorage_SmokeTest() {
-	runObjectE2ETest(suite.helper, suite.k8sh, suite.Suite, defaultRookNamespace)
+	runObjectE2ETest(suite.helper, suite.k8sh, suite.Suite, suite.namespace)
 }
 
 //Test to make sure all rook components are installed and Running
 func (suite *SmokeSuite) TestRookClusterInstallation_smokeTest() {
-	checkIfRookClusterIsInstalled(suite.k8sh, suite.Suite, defaultRookSystemNamespace, defaultRookNamespace)
+	checkIfRookClusterIsInstalled(suite.k8sh, suite.Suite, installer.SystemNamespace(suite.namespace), suite.namespace)
 }

--- a/tests/longhaul/block_test.go
+++ b/tests/longhaul/block_test.go
@@ -27,9 +27,8 @@ func TestK8sBlockLongHaul(t *testing.T) {
 }
 
 var (
-	logger               = capnslog.NewPackageLogger("github.com/rook/rook", "longhaul")
-	defaultNamespace     = "default"
-	defaultRookNamespace = "longhaul-test"
+	logger           = capnslog.NewPackageLogger("github.com/rook/rook", "longhaul")
+	defaultNamespace = "default"
 )
 
 type K8sBlockLongHaulSuite struct {
@@ -43,24 +42,25 @@ type K8sBlockLongHaulSuite struct {
 	db               *utils.MySQLHelper
 	wg               sync.WaitGroup
 	installer        *installer.InstallHelper
+	namespace        string
 }
 
 //Test set up - does the following in order
 //create pool and storage class, create a PVC, Create a MySQL app/service that uses pvc
 
 func (s *K8sBlockLongHaulSuite) SetupSuite() {
-
+	s.namespace = "longhaul-ns"
 	var err error
 	s.kh, err = utils.CreateK8sHelper(s.T)
 	assert.Nil(s.T(), err)
 
 	s.installer = installer.NewK8sRookhelper(s.kh.Clientset, s.T)
-	if !s.kh.IsRookInstalled(defaultRookNamespace) {
-		err = s.installer.InstallRookOnK8s(defaultRookNamespace)
+	if !s.kh.IsRookInstalled(s.namespace) {
+		err = s.installer.InstallRookOnK8s(s.namespace)
 		require.NoError(s.T(), err)
 	}
 
-	s.testClient, err = clients.CreateTestClient(enums.Kubernetes, s.kh, defaultRookNamespace)
+	s.testClient, err = clients.CreateTestClient(enums.Kubernetes, s.kh, s.namespace)
 	require.Nil(s.T(), err)
 
 	s.bc = s.testClient.GetBlockClient()
@@ -162,7 +162,7 @@ spec:
 	//create storage class
 	if scp, _ := s.kh.IsStorageClassPresent("rook-block"); !scp {
 		logger.Infof("Install storage class for rook block")
-		_, err = s.storageClassOperation("mysql-pool", defaultRookNamespace, "create")
+		_, err = s.storageClassOperation("mysql-pool", s.namespace, "create")
 		require.NoError(s.T(), err)
 
 		//make sure storageclass is created

